marine.c

```diff
@@ -783,6 +784,15 @@ WS_DLL_PUBLIC int init_marine(void) {
     return 0;
 }
 
+WS_DLL_PUBLIC int init_marine(void) {
+    if (marine_loaded) {
+        return -2;
+    }
+    int return_code = _init_marine();
+    marine_loaded = TRUE;
```

Comment:
Im not sure, but doesn't it shouldn't be set if `return_code` is not 0?

Reply:
Questionable.
I mean, if we failed to `_init_marine` we'd probably fail the next time we try too.
Perhaps a better name for that variable will be `can_init_marine`.

Reply:
You're both correct. I think the implementation is right, but the variable name can be improved.